### PR TITLE
fix: GH workflow test

### PR DESCRIPTION
**Pull request by mihai-sysbio**
#### Main improvements in this PR:
This PR is continuing #629 aimed to fix the GH workflow test of the import-export cycle, thus resolving #628.

**I hereby confirm that I have:**

- [ ] Tested my code on my own computer for running the model
- [x] Selected `develop` as a target branch
- [ ] Any removed reactions and metabolites have been moved to the corresponding deprecated identifier lists

### CODE REVIEW

File: code/test/testYamlConversion.m [hunk 1]
```diff
@@ -22,8 +22,12 @@
 
 
 % export to yml and then import back
-exportYaml(model,'testYamlConversion.yml');
-importedHumanGEM = importYaml('testYamlConversion.yml', true);
+try
+    exportYaml(model,'testYamlConversion.yml');
+    importedHumanGEM = importYaml('testYamlConversion.yml', true);
```
Comment (mihai-sysbio) [inline code/test/testYamlConversion.m:27]: I'm suggesting a change here to make sure the workflow fails in a way that is visible:

```suggestion
    importedHumanGEM = importYaml('testYamlConversionERROR.yml', true);
```

Comment (mihai-sysbio) [reply code/test/testYamlConversion.m]: This should obviously be reverted once we can confirm the workflow behaves as expected.

Comment (mihai-sysbio) [reply code/test/testYamlConversion.m]: Alright, it looks like the action still appears as passing when it should not.

Comment (haowang-bioinfo) [reply code/test/testYamlConversion.m]: maybe add a new action file that can be a duplicate of `Test YAML conversion` for testing?

**Comment by haowang-bioinfo** [comment]
the reported error in yaml import/export cycle is fixed by correcting typos and reverting to original subsystem value extraction (e118e1d64cc35cbaa2dc329d6bf0322cc6c298b5); while the issue with GH action remains

**Review by haowang-bioinfo — APPROVED** [review]
I tested this PR by locally running `testYamlConversion` and tandemly  import/export the yaml file, and everything is fine.

would be good to have checks for unknown keys typos in yaml, but have no idea how to reactivate the "idling" action

**Comment by mihai-sysbio** [comment]
I'm still working on getting the workflow to fail in a visible way.

File: code/test/testYamlConversion.m (hunk 1 above)
Comment (mihai-sysbio) [reply code/test/testYamlConversion.m]: > add a new action file

I'm not seeing how that would make any difference in the testing phase.

**Review by feiranl — APPROVED** [review]
LGTM!

**Comment by haowang-bioinfo** [comment]
could this be merged, how's going with the workflow test?

**Comment by mihai-sysbio** [comment]
> could this be merged, how's going with the workflow test? @mihai-sysbio

The workflow still appears as passing in the [latest workflow run](https://github.com/SysBioChalmers/Human-GEM/actions/runs/5274765201/jobs/9539549853?pr=635) even though it clearly prints out an error. I am having difficulties relaying the `exit(1)` from Matlab to bash. I've tried a few different things but none seemed to work. In the meantime, this PR is converted as to draft, since merging it would not resolve the intended issue.

**Comment by haowang-bioinfo** [comment]
> In the meantime, this PR is converted as to draft, since merging it would not resolve the intended issue.

the aim of this PR can be reduced to only resolving import/export cycle, which is a bug of develop that is now jammed

> The workflow still appears as passing in the [latest workflow run](https://github.com/SysBioChalmers/Human-GEM/actions/runs/5274765201/jobs/9539549853?pr=635) even though it clearly prints out an error. I am having difficulties relaying the exit(1) from Matlab to bash. I've tried a few different things but none seemed to work. 

yes this workflow problem can be left out and resolve later

File: code/test/testYamlConversion.m [hunk 2]
```diff
@@ -20,24 +17,24 @@
...
 
-
 % export to yml and then import back
 try
     exportYaml(model,'testYamlConversion.yml');
-    importedHumanGEM = importYaml('testYamlConversion.yml', true);
-catch
-    error('There are problems during the conversion import and export');
-end
+    importedHumanGEM = importYaml('testYamlConversionFAIL.yml', true);
```
Comment (mihai-sysbio) [inline code/test/testYamlConversion.m:23]: This purposefully introduced error should be reverted:
```suggestion
    importedHumanGEM = importYaml('testYamlConversion.yml', true);
```

**Comment by mihai-sysbio** [comment]
The workflow has been updated to as to fail in an obvious way if problems are encountered in commit 29113a1692e270b0c54e22f0d325ace870aec3da. The following commit [927fb07](https://github.com/SysBioChalmers/Human-GEM/pull/635/commits/927fb07e73f20ae630ce8ac180984d23b87793fc) removed the purposefully-introduced error, which resulted in all workflows passing. Therefore, the core issue is now resolved.

File: .github/workflows/check-metabolictasks.yml [hunk 3]
```diff
@@ -8,12 +8,11 @@ jobs:
     timeout-minutes: 60
     strategy:
       matrix:
-        task-type: [essential, verification]
+        task-type: [essential, verification, failure]
```
Comment (mihai-sysbio) [inline .github/workflows/check-metabolictasks.yml:11]: This error was purposefully introduced so as to trigger a visible error in the workflow that verifies metabolic tasks. This worked as expected for commit [visibly](https://github.com/SysBioChalmers/Human-GEM/pull/635/commits/ba6a1c7c640c407db8f9ffa3ce0698530b6f9e25)
[ba6a1c7](https://github.com/SysBioChalmers/Human-GEM/pull/635/commits/ba6a1c7c640c407db8f9ffa3ce0698530b6f9e25) and will therefore be reverted:
```suggestion
        task-type: [essential, verification]
```

**Comment by mihai-sysbio** [comment]
Due to significant changes and slight scope creep, reviews will be re-requested.

**Comment by haowang-bioinfo** [comment]
if I can re-introduce the bug happened in Yaml before and see what happens?

**Comment by mihai-sysbio** [comment]
> if I can re-introduce the bug happened in Yaml before and see what happens?

good idea, please play around this different test cases

**Comment by mihai-sysbio** [comment]
Another idea, should the `disp` function calls be replaced with `warn`?

**Comment by haowang-bioinfo** [comment]
the bug re-troduced by 97d030bee36dea37d3ff9de8a7925ecd26b40af1 suppose to lead to failure of both yaml-validation and yaml-conversion actions. But the latter still gives a pass, therefore this workflow bug persists.

**Comment by mihai-sysbio** [comment]
@haowang-bioinfo the reintroduced but caused a failure higher up in the testing code, which was not caught by the same mechanism. The previous commit [6e5d66c](https://github.com/SysBioChalmers/Human-GEM/pull/635/commits/6e5d66c402ad547a8e67b573c806ee32d5bba14a) address this.

**Comment by mihai-sysbio** [comment]
If there is any other bug you'd want to test with, please go ahead @haowang-bioinfo

**Comment by mihai-sysbio** [comment]
Oh how interesting, none of tests capture the bug introduced in [5f39c61](https://github.com/SysBioChalmers/Human-GEM/pull/635/commits/5f39c61e3e4d6e8d3bb425037755fda90a27b7ff). The actions are passing, because the tests are passing. I would be inclined to report this as separate bug - the workflows now seem to reflect the status of the tests, which was the original aim of this PR.

**Comment by haowang-bioinfo** [comment]
go ahead please

**Comment by mihai-sysbio** [comment]
@haowang-bioinfo the purposefully introduced bug has now been caught and may be removed.

**Comment by haowang-bioinfo** [comment]
with #672, both `yaml-conversion` and `yaml-validation` actions can catch the typo in MAM1261m

**Pull request merged**